### ISSUE TITLE
Section labelled by using aria-labelledby attribute is not announced when focusing inside (unless it has a "tab" element inside...)

### ISSUE THREAD

**Issue by xelloss1012** [nvaccess/nvda#13224]
### Steps to reproduce:
Have an HTML page with this code and then try to tab through the link in each section:

> &lt;section aria-labelledby="sec-1"&gt;
>   &lt;h3 id="sec-1"&gt;Section 1&lt;/h3&gt;
>   &lt;a href="#"&gt;Link 1&lt;/a&gt;
> &lt;/section&gt;
> &lt;section aria-label="Section 2"&gt;
>   &lt;h3 id="sec-2"&gt;Section 2&lt;/h3&gt;
>   &lt;a href="#"&gt;Link 2&lt;/a&gt;
> &lt;/section&gt;	
> &lt;section aria-labelledby="sec-3"&gt;
>   &lt;h3 id="sec-3"&gt;Section 3&lt;/h3&gt;
>   &lt;a href="#" role="tab"&gt;Link 3&lt;/a&gt;
> &lt;/section&gt;
> 

### Actual behavior:
The section 1, that has an aria-labelledby attribute is announced just as "region" and the text into the heading is not read.

The section 2 that has aria-label attribute with a text directly inside, is announced as "Section 2 region" (as expected).

The section 3, that has aria-labelledby attribute but inside has a link with role="tab", is announced as "Section 3 region", reading the text into the heading before the word "region" (as expected).
### Expected behavior:
In the section 1, I expected the heading linked by the aria-describedby attribute to be read (just like the sections 2 and 3), I do not understand why it does not work the same way in all 3 cases.

**Note:** I have used &lt;a&gt; elements in this example to simplify the issue, but it seems to happen also with any other element like inputs, buttons, focusable divs, etc. The only element that seems makes a difference, as far as a I have seen, is an &lt;a&gt; element with a role="tab" (maybe there are other elements or combinations, but I haven't been able to find them)... I'm not sure if it's by design or if it's some kind of coincidence.
#### NVDA installed/portable/running from source:
installed
#### NVDA version:
2021.3.1
#### Windows version:
10
#### Name and version of other software in use when reproducing the issue:
Google Chrome 96.0.4664.110
#### Other information about your system:
N/A
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes. 2021.1

Same behavior
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by feerrenrut** [comment]
[Code pen ](https://codepen.io/reefturner/pen/OJxBwKm) ([full page view](https://codepen.io/reefturner/full/OJxBwKm)) for testing.

I believe this was the intended behavior while in browse mode. The referenced element (heading 2) is visible page content, and would be read when moving with down arrow. It isn't spoken in this case to prevent being overly verbose (speaking the name once when navigating to the heading, then again when entering the region). By pressing tab you are skipping over the heading, so it isn't reported. NVDA perhaps could do something smarter here, eg speak the container (region) name when the referenced node was skipped by the last movement. This would be quite a new concept, would require resolving many edge-cases, and may be controversial.

Note that in focus mode (where the heading can not be navigated to because it isn't focusable) the region name is reported.

**Comment by xelloss1012** [comment]
Ok, this explains why it works when the first element inside is a role="tab" element, since it seems to change to focus mode. Also explains why it works when "aria-label" is attached, since this way there is no repeated content...

Just as a test, I added an "aria-hidden" to the heading element and then NVDA reads the "aria-labelledby" element, because this way the content is not repeated (I won't do in a real case though!).

I think it's a bit weird that it reads out "region landmark" and not its accesible name when using "aria-labelledby" but your explanation is consistent.

By the way, just for curiosity, I checked that this behavior happens with any landmark (section, article, main, banner, etc.) but not with other non landmark roles like "group", in this case the "aria-labelledby" element is read out and the verbosity is not prevented.

Thank you for your response!
